torch/csrc/profiler/python/init.cpp

```diff
@@ -431,6 +433,137 @@ void initPythonBindings(PyObject* module) {
     return py_symbolize(tb_ptrs);
   });
   installCapturedTracebackPython();
+  initRecordFunctionFastBindings(m.ptr());
+}
+
+/* [NOTE: RecordFunctionFast]
+ * These are an alternate way to call record_function from python.
+ * The typical context manager is slow (~14us on benchmarks in Aug 2023), which
+ * is usually fine for module-level annotations in python, but slow for per-op
+ * annotations. Part of the reason it is slow is because the calls go through
+ * the dispatcher, in order to make the record_function calls work with
+ * torchscript.
+ *
+ * This implementation doesn't go through the dispatcher and so it won't work
+ * with any feature relying on the dispatcher (e.g. torchscript or
+ * torch.compile)
+ *
+ * An alternate solution would be to implement a python context manager that
+ * calls into C++ for the enter/exit function:
+ *    @contextlib.contextmanager
+ *    def record_function_fast(name):
+ *      rf = torch._C._record_function_fast_enter(name)
+ *      try:
+ *        yield
+ *      finally:
+ *        torch._C._record_function_fast_exit(rf)
+ * The C++ implementation here is faster by ~0.2-0.4us per context manager.
+ */
+
+namespace {
+struct RecordFunctionFast {
+  PyObject_HEAD PyObject* name;
+  std::unique_ptr<at::RecordFunction> guard;
+};
+
+PyObject* RecordFunctionFast_new(
+    PyTypeObject* subtype,
+    PyObject* args,
+    PyObject* kwargs) {
+  RecordFunctionFast* self = (RecordFunctionFast*)subtype->tp_alloc(subtype, 0);
+  if (self != nullptr) {
+    self->name = nullptr;
+    self->guard.reset();
+  }
+  return (PyObject*)self;
+}
+
+int RecordFunctionFast_init(
+    PyObject* selfGeneric,
+    PyObject* args,
+    PyObject* kwargs) {
+  auto self = (RecordFunctionFast*)selfGeneric;
+  constexpr const char* kwlist[] = {"name", nullptr};
+  PyObject* name = nullptr;
+  if (!PyArg_ParseTupleAndKeywords(
+          args, kwargs, "O", const_cast<char**>(kwlist), &name)) {
+    return -1;
+  }
+  if (name) {
+    THPUtils_checkString(name);
+    self->name = name;
```

Comment:
The object returned by PyArg_ParseTupleAndKeywords is a borrowed reference. You need to incref it here and decref it during the dealloc.